contracts/Portfolio.sol

```diff
@@ -322,16 +353,34 @@ abstract contract PortfolioVirtual is Objective {
             checkPosition(args.poolId, args.owner, args.deltaLiquidity);
         if (!canUpdate) revert JitLiquidity(pool.params.jit);
 
+        (uint128 deltaAssetWad, uint128 deltaQuoteWad) =
+            (args.deltaAsset.safeCastTo128(), args.deltaQuote.safeCastTo128());
+
+        // Can only be in the case of an allocation,
+        // because pool.liquidity cannot be 0 on deallocation.
+        if (pool.liquidity == 0) {
+            // On create, virtualX and virtualY are set to initial
+            // liquidity of 1E18 and its corresponding reserves.
```

Comment:
this phrasing confused me in the beginning but I now know what you mean. it's still set to the start price per 1e18 liquidity in createPool. And then on first allocate you compute (deltaLiquidity / 1e18) * startVirtualX/Y, reset the reserves and it'll become the actual transferred in amounts.

There are a few issues that I see right now:
1. someone can just DoS any pool by allocating the initial and then deallocating it again. the pool would end up in a state with 0 liquidity and 0 virtualX/virtualY. Nobody can allocate anymore, the deltaTokens would always remain 0 as you lost the initial price?
2. I can still allocate a single deltaLiquidity unit. There are checks that deltaTokens > 0 but it could still lead to strange rounding errors that move virtualX/virtualY away from the initial computed virtualX/virtualY spot price?

One way to fix it is to always ensure that some part of deltaLiquidity remains locked up in the contract. (That's how [Uniswap also fixes a similar issue](https://github.com/Uniswap/v2-core/blob/ee547b17853e71ed4e0101ccfd52e70d5acded58/contracts/UniswapV2Pair.sol#L121).) You could ensure here in the `pool.liquidity == 0` case that `deltaLiquidty > 1e6` and then reduce the delta liquidity for the first depositor: `deltaLiquidity -= 1e6`. (`1e6` is just a value that I made up right now.) This still transfers in tokens for the full `deltaLiquidity` but ensures that they can not deallocate everything anymore.

Reply:
Agree with this fix!

Another note on this side effect: the pool reserves != physical reserves if a pool has been created but not allocated to yet.

Overall it's difficult to get around this without having users do the first allocate on `create`

Reply:
In the case liquidity is being added to a pool for the first time, a small portion of that liquidity is removed from the callers position so it is permanently burned, to avoid the scenario of all liquidity being removed from the pool: https://github.com/primitivefinance/portfolio/blob/a4b641dfcc1678c65fc0815a8bbcfed7121f3783/contracts/Portfolio.sol#L374-L377

- Adds a constant `BURNED_LIQUIDITY` equal to 1000 to burn on first allocate call.
- Adds custom error InsufficientLiquidity to throw when attempting to add less than the `BURNED_LIQUIDITY` amount on first allocate.
- Updates unit tests to consider the burned liquidity

Reply:
One note: is 1000 liquidity enough to burn or should it be more?

Reply:
From my personal experience i've seen 1e3 or 1e6, usually chosen as this value.

Reply:
Fixed here: https://github.com/primitivefinance/portfolio/commit/ee58a5de4fd39ef21b5b5a9d3bb81e7bc3721de2